### PR TITLE
refactor(clippy): Reduce some Clippy warnings

### PR DESCRIPTION
**Pull request by linyihai**
## What this PR want to do 

Tried to complete https://github.com/hyperium/hyper/issues/3802 as well as possible.

This PR wants to reduce some Clippy warnings that worth fixing.  Almost all of the changes are based on clippy prompts. And passed the MSRV 1.63 test.

Some of the warnings have not changed, mainly based on Clippy's false positives, and some clippy rules are newer than the current MSRV 1.63.

## How to test and review

Needs downgrade some crate versions to run on MSRV 1.63.
```
cargo update tokio --precise 1.38.1 
cargo update tokio-util --precise 0.7.11
cargo update hashbrown --precise 0.15.0
cargo update regex --precise 1.9.6
cargo update async-stream --precise 0.3.5
cargo update tokio-stream --precise 0.1.15
```

Run `cargo +1.63-x86_64-unknown-linux-gnu  clippy --workspace --no-deps --all-targets --features="full"  -- -D warnings`
The warnings/error reduces to 7.

### CODE REVIEW

File: src/client/conn/http1.rs [hunk 1]
```diff
@@ -302,6 +302,12 @@ where
 
 // ===== impl Builder
 
+impl Default for Builder {
```
Comment (seanmonstar) [inline src/client/conn/http1.rs:305]: Since this _adds_ something new, let's leave it as a feature PR to be separate.

File: src/proto/h1/role.rs [hunk 2]
```diff
@@ -77,7 +77,7 @@ where
         return Ok(None);
     }
 
-    let _entered = trace_span!("parse_headers");
+    trace_span!("parse_headers");
```
Comment (seanmonstar) [inline src/proto/h1/role.rs:80]: I think this changes the functionality.

File: src/server/conn/http1.rs [hunk 3]
```diff
@@ -227,6 +227,12 @@ where
 
 // ===== impl Builder =====
 
+impl Default for Builder {
```
Comment (seanmonstar) [inline src/server/conn/http1.rs:230]: Leave this for a feature PR, too.

File: tests/client.rs [hunk 4]
```diff
@@ -1827,7 +1827,7 @@ mod conn {
             sock.set_write_timeout(Some(Duration::from_secs(5)))
                 .unwrap();
             let mut buf = [0; 4096];
-            sock.read(&mut buf).expect("read 1");
+            let _ = sock.read(&mut buf).expect("read 1");
```
Comment (seanmonstar) [inline tests/client.rs:1830]: Why does clippy want this change? Seems like more noise, why can't we just ignore the integer?

**Review by seanmonstar — COMMENTED** [review]
Thanks, several of these are fine changes. I comment inline on the ones I think clippy is wrong about.

File: src/client/conn/http1.rs (hunk 1 above)
Comment (linyihai) [reply src/client/conn/http1.rs]: OK.

File: src/proto/h1/role.rs (hunk 2 above)
Comment (linyihai) [reply src/proto/h1/role.rs]: Yes, This is a Clippy false positive warnings.

File: tests/client.rs (hunk 4 above)
Comment (linyihai) [reply tests/client.rs]: My Vscode shows that it needs handle the return amount. 

Well, this isn't a very valuable change by comparison, and 'let _' is another way to ignore the return value

**Pull request merged**